qemu/tests/virtio_port_hotplug.py

```diff
@@ -31,32 +32,44 @@ def run(test, params, env):
 
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
+    os_type = params["os_type"]
     timeout = int(params.get("login_timeout", 360))
-    if params["os_type"] == "windows":
-        run_bg_test(test, params, vm)
-    for repeat in range(int(params.get("repeat_times", 1))):
-        repeat += 1
+    module = params.get("modprobe_module")
+    check_module = params.get("check_module", "yes")
+    session = vm.wait_for_login()
+    if os_type == "windows":
+        driver_name = params["driver_name"]
+        session = utils_test.qemu.windrv_check_running_verifier(
+            session, vm, test, driver_name)
+    if module and check_module == 'yes':
+        error_context.context("Load module %s" % module, logging.info)
+        session.cmd("modprobe %s" % module)
+        time.sleep(1)
+    session.close()
+
+    for port in params.objects("serials"):
         session = vm.wait_for_login(timeout=timeout)
-        module = params.get("modprobe_module")
-        if module:
-            error_context.context("Load module %s" % module, logging.info)
-            session.cmd("modprobe %s" % module)
-        for port in params.objects("serials"):
-            port_params = params.object_params(port)
-            if not port_params['serial_type'].startswith('virt'):
-                continue
-            virtio_port = vm.devices.get(port)
-            if not virtio_port:
-                test.fail("Virtio Port '%s' not found" % port)
-            chardev_qid = virtio_port.get_param("chardev")
-            try:
-                port_chardev = vm.devices.get_by_qid(chardev_qid)[0]
-            except IndexError:
-                test.error("Failed to get device %s" % chardev_qid)
-            if module:
+        port_params = params.object_params(port)
+        if not port_params['serial_type'].startswith('virt'):
+            continue
+        virtio_port = vm.devices.get(port)
+        if not virtio_port:
+            test.fail("Virtio Port '%s' not found" % port)
+        chardev_qid = virtio_port.get_param("chardev")
+        try:
+            port_chardev = vm.devices.get_by_qid(chardev_qid)[0]
+        except IndexError:
+            test.error("Failed to get device %s" % chardev_qid)
+        if port_params['serial_type'] == 'virtserialport':
+            params['file_transfer_serial_port'] = port
+            run_bg_test(test, params, vm)
+        for repeat in range(int(params.get("repeat_times", 1))):
```

Comment:
params.get_numeric("repeat_times", 1)

Reply:
Thanks and update.

Reply:
Please provider your latest test results. code looks OK to me.

Reply:
Thanks, there is a little problem when running `run_bg_test` with Linux for transferring data too fast, will update result later.